### PR TITLE
MES-3437: Prevent rekeying of a submitted test

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
## Description

- Makes an extra request to the search completed tests endpoint preceding the search for booking request to prevent submitted tests being rekeyed.
- Renames method to distinguish between tests and bookings.
- Adds test coverage.

Link to story - https://jira.dvsacloud.uk/browse/MES-3437

## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

**Review by rossfauldsbjss — APPROVED** [review]
Looks good. Just would like to see how it works if a rekey is attempted 
when offline. So long as that checks out, it's good to go.

File: src/pages/rekey-search/rekey-search.effects.ts [hunk 1]
```diff
@@ -2,33 +2,51 @@ import { Injectable } from '@angular/core';
...
+        }),
         catchError((err: any) => {
-          return of(new rekeySearchActions.SearchBookedTestFailure(err));
+          const rekeySearchParams = {
+            applicationReference: action.appRef,
+            staffNumber: action.staffNumber,
+          };
+          return this.rekeySearchProvider.getBooking(rekeySearchParams).pipe(
+            map(response => this.compressionProvider.extractTestSlotResult(response.toString())),
+            map((testSlot: any) => new SearchBookedTestSuccess(testSlot, action.staffNumber)),
```
Comment (rstepnowski) [inline src/pages/rekey-search/rekey-search.effects.ts]: I'm having mixed feelings about this success scenario being dealt with within a `catchError` block :P
Not sure atm if there is a better way to do it though :)

Comment (ammarhaiderbjss) [reply src/pages/rekey-search/rekey-search.effects.ts]: I could make it more specific to be fair, looking out for the specific status code that denotes there is no submitted test.

Comment (ammarhaiderbjss) [reply src/pages/rekey-search/rekey-search.effects.ts]: But yeah not sure there's a way around the catchError :(

Comment (ammarhaiderbjss) [reply src/pages/rekey-search/rekey-search.effects.ts]: Have made some amendments if you'd like to re-review @rstepnowski

**Review by TomBillingtonUK — APPROVED** [review]

**Review by rstepnowski — APPROVED** [review]

**Pull request merged**